overlord/configstate/config/helpers.go

```diff
@@ -119,6 +120,96 @@ func GetFromChange(snapName string, subkeys []string, pos int, config map[string
 	return GetFromChange(snapName, subkeys, pos+1, configm, result)
 }
 
+// StoreConfigSnapshotMaybe makes a copy of config -> snapSnape configuration into the versioned config.
+// It doesn't do anything if there is no configuration for given snap in the state.
+// The caller is responsible for locking the state.
+func StoreConfigSnapshotMaybe(st *state.State, snapName string, rev snap.Revision) error {
```

Comment:
I don't think that's necessary. All of the configuration logic handles missing the same as an empty configuration, and considers that okay.